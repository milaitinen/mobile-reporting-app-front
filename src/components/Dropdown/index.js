import Dropdown from './Dropdown';
import NestedDropDown from './NestedDropdown';

export { Dropdown, NestedDropDown };