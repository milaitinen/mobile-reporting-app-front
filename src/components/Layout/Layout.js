import React, { Component } from 'react';
import { StyleSheet, View, Animated } from 'react-native';
import { ListItem } from 'react-native-elements';

class Layout extends Component{
    constructor(props){
        super(props);
        this.state = {
            title      : props.title,           // Title which the layout inherits from TemplateScreen.
            nofForms   : props.nofForms,        // Number of forms which the layout inherits from TemplateScreen.
            layoutID   : props.layoutID,        // The specific layoutID which the layout inherits from TemplateScreen.
            expanded   : false,                 // Checks whether the forms of the layout are shown or not.
            animation  : new Animated.Value(50) /* Initializes the animation state as 50 (same height as the ListItem
                                                   component which includes the title of the Layout etc.)
                                                   This is the minimum height when the layout isn't expanded. */
        };
    }

    // Toggle function for closing and expanding the layout.

    toggle(){
        const
            initialValue = this.state.expanded? this.state.maxHeight + this.state.minHeight : this.state.minHeight,
            finalValue   = this.state.expanded? this.state.minHeight : this.state.maxHeight + this.state.minHeight;

        this.setState({
            expanded : !this.state.expanded
        });
        // Animation for closing and opening
        this.state.animation.setValue(initialValue);
        Animated.spring(
            this.state.animation,
            {
                toValue: finalValue
            }
        ).start();
    }
    // Sets maximum height when opened.
    _setMaxHeight(event){
        this.setState({
            maxHeight   : event.nativeEvent.layout.height
        });
    }
    // Sets minimum height when closed.
    _setMinHeight(event){
        this.setState({
            minHeight   : event.nativeEvent.layout.height
        });
    }

    // Calls the inherited createNew function which is explained in TemplateScreen class.

    createNew(layoutID) {
        this.props.createNew(layoutID);
    }

    render(){
        /* Renders the Layout and its children, which are defined in the TemplateScreen class.
           The TemplateScreen uses FlatList component as the Layout components child.
         */
        return (
            <Animated.View
                style={[styles.container,{ height: this.state.animation }]}>
                <View onLayout={this._setMinHeight.bind(this)}>
                    <ListItem
                        containerStyle={ styles.ListItemStyle }
                        onPress={this.toggle.bind(this)} // Opens or closes the layout.
                        title={this.state.title} // Title of the layout.
                        subtitle={this.state.nofForms + ' Forms'} // Number of forms as a subtitle.
                        rightIcon={{ name: 'arrow-right', type: 'font-awesome', style: { marginRight: 10, fontSize: 15 } }}
                        onPressRightIcon={() => this.createNew(this.state.layoutID)} /* Navigates to NewReportScreen when
                                                                                        pressed.*/
                    />

                </View>

                <View style={styles.body} onLayout={this._setMaxHeight.bind(this)}>
                    {this.props.children}
                </View>

            </Animated.View>
        );
    }
}

const styles = StyleSheet.create({

    ListItemStyle: {
        height: 50
    },
    container: {
        backgroundColor: '#fff',
        margin:10,
        overflow:'hidden'
    },
    body: {
        padding     : 10,
        paddingTop  : 0
    }
});

export default Layout;
