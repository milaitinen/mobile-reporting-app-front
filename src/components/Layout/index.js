import Layout from './Layout';
//import styles from './styles';

export { Layout };
