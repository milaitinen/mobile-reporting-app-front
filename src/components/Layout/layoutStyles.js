import { StyleSheet } from 'react-native';

const layoutStyles = StyleSheet.create({
    ListItemTitleStyle: {
        height: 60,
        borderRadius: 5,
        backgroundColor: '#fff',
        paddingTop: 0,
        paddingRight: 0,
        paddingLeft: 1,
        paddingVertical: 0,
        paddingBottom: 0,
    },

    ListItemStyle: {
        height: 60,
        borderBottomColor: '#b4b4b470',

    },

    container: {
        backgroundColor: '#e0e8eb',
        margin: 10,
        overflow:'hidden',
        borderRadius: 5,
        height: 60,
        elevation: 3,
        shadowOffset: {
            width: 0,
            height: 3,
        }
    },

    titleContainer: {
        flexDirection: 'row',
        paddingRight: 2,
        paddingLeft: 0,

        //paddingTop: 10,
    },

    titleStyle: {
        paddingTop: 5,
        fontFamily: 'Roboto-Light',
        fontSize: 18,
    },

    listTitleStyle :{
        fontFamily: 'Roboto-Light',
        fontSize: 16,
        marginRight: 0,
    },

    rightIconStyle: {
        marginRight: 0,
        marginTop: 0,
        marginBottom: 0,
        fontSize: 50,
        color: '#77a482',
        backgroundColor: '#99d9ad',
        paddingLeft: 8,
        paddingRight: 5,
        paddingTop: 5,
        paddingBottom: 6,
        borderRadius: 5,
        //borderColor: '#000',
        //borderWidth: 1
    },

    leftIconStyle: {
        marginLeft: 0,
        fontSize: 50,
        color: '#b4b4b4',
        paddingRight: 0,
    },

    badgeContainerStyleA: {
        backgroundColor: '#77a482',
        elevation: 5,
        borderRadius: 5,
    },

    badgeContainerStyleP: {
        backgroundColor: '#f3fe99',
        elevation: 3,
        shadowOffset: {
            width: 0,
            height: 3,
        },
        borderRadius: 5,
        //paddingRight: 3,
        //paddingLeft: 3,
    },

    badgeTextStyle: {
        color: '#666666',
        padding: 1
    },

    body: {
        padding: 10,
        paddingTop: 0,
    },

    more: {
        color: '#5088b8',
        textDecorationLine: 'underline',
        padding: 10,
        textAlign: 'center',
    },
});

export default layoutStyles;