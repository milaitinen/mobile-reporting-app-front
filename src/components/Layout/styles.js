import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({

    animatedContainer: {
        backgroundColor: '#e0e8eb',
        marginLeft:0,
        marginRight: 0,
        overflow:'hidden',
        borderRadius: '$containerBorderRadius',
        height: 60,
        elevation: 3,
        shadowOffset: {
            width: 0,
            height: 3,
        },
        '@media (max-width: 350)': {
            height: 50,
            margin: 8,
        },
        '@media (min-width: 350)': {
            height: 60,
            margin: 10,
        }
    },

    templateContainer: {
        borderRadius: 5,
        flexDirection: 'column',
        backgroundColor: '$primaryWhite',
        paddingTop: 0,
        paddingRight: 0,
        paddingLeft: 0,
        paddingVertical: 0,
        paddingBottom: 0,
        '@media (max-width: 350)': {
            height: 50,
        },
        '@media (min-width: 350)': {
            height: 60,
        }
    },

    subtitle: {
        flexDirection: 'column',
    },

    folderIcon: {
        marginLeft: 0,
        color: '#b4b4b4',
        paddingRight: 0,
        '@media (max-width: 350)': {
            fontSize: 40,
        },
        '@media (min-width: 350)': {
            fontSize: 50,
        }
    },

    reportListContainer: {
        paddingTop: 0,
        '@media (max-width: 350)': {
            padding: 7,
        },
        '@media (min-width: 350)': {
            padding: 10,
        }
    },

    reportContainer: {
        borderBottomColor: '#b4b4b470',

        '@media (max-width: 350)': {
            height: 60,
        },
        '@media (min-width: 350)': {
            height: 60,
        }

    },

    reportTitle: {
        '@media (max-width: 350)': {
            fontSize: 12
        } ,
    },

    more: {
        color: '#6da7e7',
        textDecorationLine: 'underline',
        padding: 10,
        textAlign: 'center',

        '@media (max-width: 350)': {
            padding: 7,
            fontSize: 12,
        },
        '@media (min-width: 350)': {
            padding: 10,
        },
    },

    noMoreReports: {
        color: '#a9a9a9',
        padding: 10,
        textAlign: 'center',

        '@media (max-width: 350)': {
            padding: 7,
            fontSize: 12,
        },
        '@media (min-width: 350)': {
            padding: 10,
        },
    },
});