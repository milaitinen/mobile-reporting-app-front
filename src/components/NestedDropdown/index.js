import NestedDropdown from './NestedDropdown';

export { NestedDropdown };