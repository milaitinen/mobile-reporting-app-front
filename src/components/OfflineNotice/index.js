import OfflineNotice from './OfflineNotice';

export { OfflineNotice };