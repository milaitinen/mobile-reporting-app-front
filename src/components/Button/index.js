import SignInButton from './SignInButton';
import Button from './Button';
import styles from './signInstyles';

export { Button, SignInButton, styles };