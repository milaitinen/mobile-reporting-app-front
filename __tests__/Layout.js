import 'react-native';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import { Layout } from '../src/components/Layout';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
    const tree = renderer.create(
        <Layout data={[]} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

describe('Layout', () => {

    const wrapper = renderer.create(<Layout data={[]} />);   // render React components to pure JavaScript objects
    const inst = wrapper.getInstance();                      // Return the instance corresponding to the root element, if available.

    describe('toggleExpanded()', () => {
        it('should set state "expanded" to true when called once', () => {
            inst.toggleExpanded();
            expect(inst.state.expanded).toBe(true);
        });
        it('should set state "expanded" to false when called twice', () => {
            inst.toggleExpanded();
            expect(inst.state.expanded).toBe(false);
        });
    });

    describe('showMore()', () => {
        it('should update the states itemsCount, updated, and maxHeight, when called', () => {
            expect(inst.state.itemsCount).toBe(20);
            expect(inst.state.updated).toBe(false);
            expect(inst.state.maxHeight).toBe(475);

            inst.showMore();

            expect(inst.state.itemsCount).toBe(40);
            expect(inst.state.updated).toBe(true);
        });
    });
});




