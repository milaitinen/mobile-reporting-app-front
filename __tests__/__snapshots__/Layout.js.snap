// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "height": 60,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "#bbb",
              "borderBottomWidth": 1,
              "paddingBottom": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            undefined,
            false,
            false,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "marginLeft": 10,
            },
            undefined,
          ]
        }
      >
        <View
          disabled={false}
          onLongPress={undefined}
          onPress={undefined}
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          underlayColor="white"
        >
          <View>
            <View
              onPress={undefined}
              style={
                Array [
                  false,
                  false,
                  false,
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
              underlayColor="white"
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#bdc6cf",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Array [
                        Object {
                          "marginRight": 8,
                        },
                        undefined,
                      ],
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={undefined}
          >
            <View />
          </View>
          <View
            style={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 15,
                      "fontWeight": "600",
                      "marginTop": 1,
                    },
                    false,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              undefined Reports
            </Text>
          </View>
        </View>
        <View
          style={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View>
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Array [
                          undefined,
                          Object {
                            "backgroundColor": "#A4CBE7",
                            "color": "#8096a8",
                          },
                        ],
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View>
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "black",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Array [
                          undefined,
                          Object {
                            "backgroundColor": "#99d9ad",
                            "color": "#77a482",
                          },
                        ],
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={undefined}
  >
    <RCTScrollView
      ListFooterComponent={null}
      data={Array []}
      disableVirtualization={false}
      extraData={5}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
